src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -359,6 +358,10 @@ private void UpdateWorkingDirectory()
                     _solutionManager,
                     _deleteOnRestartManager);
 
+                var packagesConfigInstalled = new Dictionary<NuGetFramework, HashSet<PackageIdentity>>();
```

Comment:
The framework -> package pair needs to be tracked so that it resolve the dependencies against that framework. The reason for storing them all up front is so this can be done at the end by ordering on the framework instead of just taking the first project as VS returns it.